Source/DafnyCore/Verifier/Translator.ClassMembers.cs

```diff
@@ -1069,6 +1070,18 @@ public partial class Translator {
         layer = new Boogie.IdentifierExpr(f.tok, "$LZ", predef.LayerType); // $LZ
       }
 
+      if (f.IsOpaque) {
+        Contract.Assert(overridingFunction.IsOpaque);
+        var revealVar = new Boogie.BoundVariable(f.tok, new Boogie.TypedIdent(f.tok, "reveal", Boogie.Type.Bool));
+        forallFormals.Add(revealVar);
+        reveal = new Boogie.IdentifierExpr(f.tok, revealVar);
+        argsJF.Add(reveal);
+      } else if (overridingFunction.IsOpaque) {
+        // We can't use a bound variable $fuel, because then one of the triggers won't be mentioning this $fuel.
+        // Instead, we do the next best thing: use the literal false.
+        reveal = new Boogie.IdentifierExpr(f.tok, "false", Boogie.Type.Bool);
```

Comment:
Can you create a literal expression instead of an identifier expression here? I think this accidentally works but is confusing, since `false` is an actual value rather than a name for a value, at least in the semantics I understand.